### PR TITLE
Use Lazy loading instead of Eager to reduce database calls

### PR DESCRIPTION
**Pull request by Ben-Edwards-cgi**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by EPatterson1 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/juror/api/moj/repository/jurorresponse/IJurorCommonResponseRepositoryModImpl.java [hunk 1]
```diff
@@ -38,6 +38,7 @@ public List<Tuple> getJurorResponseDetailsByUsernameAndStatus(String staffLogin,
             .where(QCombinedJurorResponse.combinedJurorResponse.processingStatus.in(processingStatus))
             .where(QCombinedJurorResponse.combinedJurorResponse.juror.bureauTransferDate.isNull())
             .where(QJurorPool.jurorPool.isActive.isTrue())
+            .where(QCombinedJurorResponse.combinedJurorResponse.juror.bureauTransferDate.isNull())
```
Comment (akikrahman1) [inline src/main/java/uk/gov/hmcts/juror/api/moj/repository/jurorresponse/IJurorCommonResponseRepositoryModImpl.java:41]: is this a duplicate line?

File: src/main/java/uk/gov/hmcts/juror/api/moj/domain/User.java [hunk 2]
```diff
@@ -76,7 +76,7 @@ public class User implements Serializable {
     @Enumerated(EnumType.STRING)
     private UserType userType;
 
-    @ElementCollection(fetch = FetchType.EAGER)
```
Comment (akikrahman1) [inline src/main/java/uk/gov/hmcts/juror/api/moj/domain/User.java:79]: was there a reason this was eager?

File: src/main/java/uk/gov/hmcts/juror/api/moj/repository/jurorresponse/IJurorCommonResponseRepositoryModImpl.java (hunk 1 above)
Comment (Ben-Edwards-cgi) [reply src/main/java/uk/gov/hmcts/juror/api/moj/repository/jurorresponse/IJurorCommonResponseRepositoryModImpl.java:41]: Good spot updated

File: src/main/java/uk/gov/hmcts/juror/api/moj/domain/User.java (hunk 2 above)
Comment (Ben-Edwards-cgi) [reply src/main/java/uk/gov/hmcts/juror/api/moj/domain/User.java:79]: When the User object was created it was assume when ever you grab the user you will also want to grab the roles. As we integrated this more throughout the application it turns out this was not the case so we no longer need it to EAGER load

**Review by akikrahman1 — APPROVED** [review]

**Pull request merged**